src/interfaces/coral_web/src/hooks/chat.ts

```diff
@@ -248,9 +252,59 @@ export const useChat = (config?: { onSend?: (msg: string) => void }) => {
               break;
             }
 
-            case StreamEvent.TOOL_INPUT: {
-              const data = eventData.data as StreamToolInput;
-              toolEvents.push(data);
+            case StreamEvent.TOOL_CALLS_CHUNK: {
+              const data = eventData.data as StreamToolCallsChunk;
+
+              // Initiate an empty tool event if one doesn't already exist at the current index
+              const toolEvent: StreamToolCallsGeneration = toolEvents[currentToolEventIndex] ?? {
+                text: '',
+                tool_calls: [],
+              };
+              toolEvent.text += data?.text ?? '';
+
+              // A tool call needs to be added/updated if a tool call delta is present in the event
+              if (data?.tool_call_delta) {
+                const currentToolCallsIndex = data.tool_call_delta.index ?? 0;
+                let toolCall = toolEvent.tool_calls?.[currentToolCallsIndex];
+                if (!toolCall) {
+                  toolCall = {
+                    name: '',
+                    parameters: {},
+                  };
+                  toolCallParamaterStr = '';
+                }
+
+                if (data?.tool_call_delta?.name) {
+                  toolCall.name = data.tool_call_delta.name;
+                }
+                if (data?.tool_call_delta?.parameters) {
+                  toolCallParamaterStr += data?.tool_call_delta?.parameters;
+
+                  // Attempt to parse the partial parameter string as valid JSON to show that the parameters
+                  // are streaming in. To make the partial JSON string valid JSON after the object key comes in,
+                  // we naively try to add `"}` to the end.
```

Comment:
This is a bit janky but unfortunately we don't receive a structured object like we do in Coral i.e. 
```
data: {"event": "tool-calls-chunk", "data": {"tool_call_delta": {"name": null, "index": 0, "parameters": "{\n            \""}, "text": ""}}

data: {"event": "tool-calls-chunk", "data": {"tool_call_delta": {"name": null, "index": 0, "parameters": "query"}, "text": ""}}

data: {"event": "tool-calls-chunk", "data": {"tool_call_delta": {"name": null, "index": 0, "parameters": "\":"}, "text": ""}}

data: {"event": "tool-calls-chunk", "data": {"tool_call_delta": {"name": null, "index": 0, "parameters": " \""}, "text": ""}}

data: {"event": "tool-calls-chunk", "data": {"tool_call_delta": {"name": null, "index": 0, "parameters": "average"}, "text": ""}}
```

vs

```
{
  "result": {
    "eventType": "STREAM_EVENT_TYPE_TOOL_CALLS_CHUNK",
    "isFinished": false,
    "toolCallsChunkStreamEvent": {
      "toolCallDelta": {
        "parameter": {
          "name": "query"
        }
      }
    }
  }
}{
  "result": {
    "eventType": "STREAM_EVENT_TYPE_TOOL_CALLS_CHUNK",
    "isFinished": false,
    "toolCallsChunkStreamEvent": {
      "toolCallDelta": {
        "parameter": {
          "name": "query",
          "value": "\""
        }
      }
    }
  }
}{
  "result": {
    "eventType": "STREAM_EVENT_TYPE_TOOL_CALLS_CHUNK",
    "isFinished": false,
    "toolCallsChunkStreamEvent": {
      "toolCallDelta": {
        "parameter": {
          "name": "query",
          "value": "average"
        }
      }
    }
  }
}
```

Reply:
:')